src/util.js

```diff
@@ -311,3 +311,16 @@ function setLocEnd(node, index) {
   }
 }
 util.setLocEnd = setLocEnd;
+
+// http://stackoverflow.com/a/7124052
+function htmlEscapeInsideDoubleQuote(str) {
+  return str
+    .replace(/&/g, '&amp;')
```

Comment:
JSX was supposed to be HTML in JS so it borrowed the HTML escaping format for strings literals. This is somethint that we are not happy about and if we are ever doing jsx2 we'll fix this. Until then we have to live with it